python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -284,6 +283,8 @@ def predict(self, X, output_class=True, algo='auto', threshold=0.5,
         return preds
 
     def predict_using_fil(self, X, delayed, **kwargs):
+        assert self.num_classes <= 2, "multi-class classification not "\
```

Comment:
why did you move the check for num classes from the main predict function to `predict_using_fil`
?

Reply:
that covers both the `score()` and `predict()` functions, and is semantically more stable - whatever the logic changes in `score()` and `predict()`, the assert will be checked only and always on GPU predictions.